hazelcast/src/main/java/com/hazelcast/security/ICredentialsFactory.java

```diff
@@ -50,6 +51,16 @@ default void init(Properties properties) {
      */
     Credentials newCredentials();
 
+    /**
+     * Creates new {@link Credentials} object for given target {@link Address}.
+     *
+     * @param address Target {@link Address} (may be {@code null})
+     * @return
+     */
+    default Credentials newCredentials(Address address) {
```

Comment:
Is this change related with the purpose of the PR? Probably, it's used somewhere, but I'd like to double check.

Reply:
Ah, nice catch. This change probably sneaked in from my Kerberos related work.